kornia/augmentation/_2d/intensity/clahe.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any, Dict, Optional, Tuple
+
+from kornia.augmentation import random_generator as rg
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.core import Tensor
+from kornia.enhance import equalize_clahe
+
+
+class RandomClahe(IntensityAugmentationBase2D):
+    r"""Apply CLAHE equalization on the input tensor randomly.
+
+    .. image:: _static/img/equalize_clahe.png
+
+    Args:
+        clip_limit: threshold value for contrast limiting. If 0 clipping is disabled.
+        grid_size: number of tiles to be cropped in each direction (GH, GW).
+        slow_and_differentiable: flag to select implementation
+        same_on_batch: apply the same transformation across the batch.
+        p: probability of applying the transformation.
+        keepdim: whether to keep the output shape the same as input (True) or broadcast it
+                 to the batch form (False).
+    .. note::
+        This function internally uses :func:`kornia.enhance.equalize_clahe`.
+
+    Examples:
+        >>> img = torch.rand(1, 10, 20)
+        >>> inv = RandomClahe()
+        >>> res = inv(img)
+        >>> res.shape
+        torch.Size([1, 10, 20])
+
+        >>> img = torch.rand(2, 3, 10, 20)
+        >>> inv = RandomClahe()
+        >>> res = inv(img)
+        >>> res.shape
+        torch.Size([2, 3, 10, 20])
+
+    To apply the exact augmenation again, you may take the advantage of the previous parameter state:
+        >>> input = torch.randn(1, 3, 32, 32)
+        >>> aug = RandomClahe(p=1.)
+        >>> (aug(input) == aug(input, params=aug._params)).all()
+        tensor(True)
+    """
+
+    def __init__(
+        self,
+        clip_limit: Tuple[float, float] = (40.0, 40.0),
```

Comment:
```suggestion
        clip_limit: tuple[float, float] = (40.0, 40.0),
```
and `from __future__ import annotations`

Reply:
@michael-2956 needs to add the import